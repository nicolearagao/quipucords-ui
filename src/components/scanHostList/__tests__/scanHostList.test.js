import React from 'react';
import { ScanHostList } from '../scanHostList';

describe('ScanHostList Component', () => {
  it('should render a basic component', () => {
    const props = {
      id: 1,
      useConnectionResults: true,
      useInspectionResults: true,
      hostsList: [
        {
          credentialName: 'dolor',
          jobType: 'connection',
          name: 'lorem',
          sourceId: 15,
          sourceName: 'lorem source'
        },
        {
          credentialName: 'set',
          jobType: 'inspection',
          name: 'ipsum',
          sourceId: 16,
          sourceName: 'ipsum source'
        }
      ]
    };

    const component = renderComponent(<ScanHostList {...props}>{({ host }) => JSON.stringify(host)}</ScanHostList>);
    expect(component).toMatchSnapshot('basic');
  });

  it('should render a component error', () => {
    const props = {
      id: 1,
      error: true,
      errorMessage: 'Lorem Ipsum.'
    };

    const component = renderComponent(<ScanHostList {...props}>{({ host }) => JSON.stringify(host)}</ScanHostList>);
    expect(component).toMatchSnapshot('error');
  });

  it('should render a component pending', () => {
    const props = {
      id: 1,
      pending: true
    };

    const component = renderComponent(<ScanHostList {...props}>{({ host }) => JSON.stringify(host)}</ScanHostList>);
    expect(component).toMatchSnapshot('pending');
  });
});
