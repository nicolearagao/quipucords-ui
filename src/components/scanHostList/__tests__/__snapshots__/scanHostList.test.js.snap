// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanHostList Component should render a basic component: basic 1`] = `
<scanhostlist
  props="{
  "id": 1,
  "useConnectionResults": true,
  "useInspectionResults": true,
  "hostsList": [
    {
      "credentialName": "dolor",
      "jobType": "connection",
      "name": "lorem",
      "sourceId": 15,
      "sourceName": "lorem source"
    },
    {
      "credentialName": "set",
      "jobType": "inspection",
      "name": "ipsum",
      "sourceId": 16,
      "sourceName": "ipsum source"
    }
  ],
  "children": [],
  "error": false,
  "errorMessage": null,
  "filter": {},
  "isMoreResults": false,
  "pending": false
}"
>
  <ul
    class="pf-c-list pf-m-plain quipucords-infinite-results quipucords-infinite-list"
  >
    <li
      class=""
    >
      {"credentialName":"dolor","jobType":"connection","name":"lorem","sourceId":15,"sourceName":"lorem source"}
    </li>
    <li
      class=""
    >
      {"credentialName":"set","jobType":"inspection","name":"ipsum","sourceId":16,"sourceName":"ipsum source"}
    </li>
  </ul>
</scanhostlist>
`;

exports[`ScanHostList Component should render a component error: error 1`] = `
<scanhostlist
  props="{
  "id": 1,
  "error": true,
  "errorMessage": "Lorem Ipsum.",
  "children": [],
  "filter": {},
  "hostsList": [],
  "isMoreResults": false,
  "pending": false,
  "useConnectionResults": false,
  "useInspectionResults": false
}"
>
  <div
    class="pf-c-empty-state quipucords-empty-state__alert"
  >
    <div
      class="pf-c-empty-state__content"
    >
      <div
        aria-label="Danger Alert"
        class="pf-c-alert pf-m-inline pf-m-plain pf-m-danger"
        data-ouia-component-id="OUIA-Generated-Alert-danger-1"
        data-ouia-component-type="PF4/Alert"
        data-ouia-safe="true"
      >
        <div
          class="pf-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            />
          </svg>
        </div>
        <h4
          class="pf-c-alert__title"
        >
          <span
            class="pf-u-screen-reader"
          >
            Danger alert:
          </span>
          t(view.error, {"context":"scan-hosts"})
        </h4>
        <div
          class="pf-c-alert__description"
        >
          t(view.error-message, {"context":"scan-hosts","message":"Lorem Ipsum."})
        </div>
      </div>
    </div>
  </div>
</scanhostlist>
`;

exports[`ScanHostList Component should render a component pending: pending 1`] = `
<scanhostlist
  props="{
  "id": 1,
  "pending": true,
  "children": [],
  "error": false,
  "errorMessage": null,
  "filter": {},
  "hostsList": [],
  "isMoreResults": false,
  "useConnectionResults": false,
  "useInspectionResults": false
}"
>
  <div
    class="pf-c-empty-state pf-m-lg quipucords-empty-state"
  >
    <div
      class="pf-c-empty-state__content"
    >
      <svg
        aria-label="Contents"
        aria-valuetext="Loading..."
        class="pf-c-spinner pf-m-sm"
        role="progressbar"
        viewBox="0 0 100 100"
      >
        <circle
          class="pf-c-spinner__path"
          cx="50"
          cy="50"
          fill="none"
          r="45"
        />
      </svg>
       
      t(view.loading)
    </div>
  </div>
</scanhostlist>
`;
